index.ts

```diff
@@ -130,146 +136,157 @@ if (syncRepoLabels) {
 const payload = require(process.env.GITHUB_EVENT_PATH as string)
 const number = (payload.issue || payload.pull_request || payload).number
 const action: string = payload.action
+const issue: Issue = payload.issue
 
-// retrieve issue by owner, repo and number from octokit
-gitHubSource
-    .getIssue(number)
-    .then(response => {
-        // Retrieved issue
-        const issue: Issue = response.data
-        const labels: string[] = [...new Set(issue.labels.map(label => label.name).concat(additionalIssueLabels))]
+const labels: string[] = [...new Set(issue.labels.map(label => label.name).concat(additionalIssueLabels))]
+// If flag for only syncing labelled issues is set, check if issue has label of specified sync type
+const skipSync = ONLY_SYNC_ON_LABEL && !issue.labels.find(label => label.name === ONLY_SYNC_ON_LABEL)
```

Comment:
not even sure whether it would be needed.
We explicitly define in the GitHub Action workflow to skip those that don't contain the label.
Checking it in the action itself would potentially cost a lot of money as GitHub rounds up to every full minute.

We can keep it but could also say that the Action should only be used in conjunction with the if contains filter within the workflow.

Reply:
Just keeping it for the backwards compatibility with the original action.
I agree that we should be using the `if` filtering on the action level though.